www/js/lib/zimfile.js

```diff
@@ -187,9 +193,21 @@ define(['xzdec_wrapper', 'zstddec_wrapper', 'util', 'utf8', 'zimDirEntry', 'file
             var pos = dirEntry.redirect ? 12 : 16;
             if (data.subarray) {
                 dirEntry.url = utf8.parse(data.subarray(pos), true);
-                while (data[pos] !== 0)
-                    pos++;
-                dirEntry.title = utf8.parse(data.subarray(pos + 1), true);
+                for (pos; pos <= MAX_SUPPORTED_DIRENTRY_SIZE; pos++) {
+                    if (data[pos] === 0) break;
+                }
+                // DEBUG
+                if (pos > 2048)  console.debug('Found dirEntry.url of size > 2KB!' + (!data[pos] ? ' (' + pos + ')' : ''), dirEntry.url);
```

Comment:
This was just diagnostic code for testing this PR, so you could see the dirEntries that would have caused a hang. I intended to remove these debug lines. But it may be useful just to comment them out, for ease of future testing? In any case, happy to change the number to the constant.

Reply:
Actually, thinking about it, if we change the number to MAX_SUPPORTED... then we are just replicating the console warning given below. So I think my original plan of deleting these lines is best. They were just there to diagnose and be able to "reproduce" the issue.